src/pybind/mgr/volumes/fs/async_job.py

```diff
@@ -174,7 +174,6 @@ def shutdown(self):
         self.cancel_all_jobs()
         with lock_timeout_log(self.lock):
             self.cv.notifyAll()
-        self.join()
```

Comment:
This is supposed to be invoked from another thread. What issue/bug are you you running into?

Reply:
On a second look, this patch is incorrect since `AsyncJob.shutdown()` and `AsyncJob.run()` will run on separate threads.